taichi/codegen/spirv/spirv_codegen.cpp

```diff
@@ -564,8 +564,31 @@ class TaskCodegen : public IRVisitor {
     {
       const int num_indices = stmt->indices.size();
       std::vector<std::string> size_var_names;
+      const auto &element_shape = stmt->element_shape;
+      std::vector<std::string> element_shape_size_var_names;
+      enum ExternalArrayLayout { layout_AOS = 0, layout_SOA = 1 };
```

Comment:
Btw I moved this enum to a common header in https://github.com/taichi-dev/taichi/pull/5066 ;) but it can be updated in this PR or followup depends on which one goes in first ;)

Reply:
I would remove all the guess works when element shape in C++ ndarray is ready :)